views/js/runner/testStore.js

```diff
@@ -0,0 +1,295 @@
+     * Database name prefix (suffixed by the test identifier)
+     * to check if we use the legacy mode (multiple dbs)
+     * or the new mode (one db per test).
+     * @type {String[]}
+     */
+    var legacyPrefixes = [
+        'actions-', 'duration-', 'test-probe', 'timer-'
+    ];
+    /**
+     * Check and select the store mode.
+     * If any of the "legacyPrefixes" store is found, we used the "multiple/legacy" mode
+     * otherwise we use the unified mode.
+     * @param {String} testId
+     * @param {Object} [preselectedBackend] - the storage backend
+     * @returns {Promise<Boolean>} true means unified
+     */
+    var selectStoreMode = function selectStoreMode(testId, preselectedBackend){
+        return store
+            .getAll(function validate(storeName){
+                return _.some(legacyPrefixes, function(prefix){
+                    return !_.isEmpty(storeName) && prefix + testId === storeName;
+                });
+            }, preselectedBackend)
+            .then(function(foundStores){
+                return _.isArray(foundStores) && foundStores.length === 0;
+            });
+    };
+
+
+    /**
+     * Get the store for the given test
+     *
+     * @param {String} testId - unique test instance id
+     * @returns {testStore} a 'wrapped' store instance
+     * @param {Object} [preselectedBackend] - the storage backend (automatically selected by default)
+     * @throws {TypeError} with no testId
+     */
+    return function testStoreLoader(testId, preselectedBackend){
+
+        var storeNames = [];
+        var volatiles = [];
+        var unifiedStore;
+
+        var getMode = Promise.resolve();
+        if(_.isUndefined(unifiedStore)){
+            getMode = selectStoreMode(testId, preselectedBackend).then(function(result){
+                unifiedStore = !!result;
+
+                logger.debug('Test store mode ' + (unifiedStore ? 'unified' : 'multiple (legacy)') + ' for ' + testId);
+            });
+        }
+
+        if(_.isEmpty(testId)){
+            throw new TypeError('The store must be identified with a unique test identifier');
+        }
+
+        /**
+         * Wraps a store and add the support of "volatile" storages
+         * @typedef {Object} testStore
+         */
+        return {
+
+            /**
+             * Get a wrapped store instance, that let's you use multiple stores inside one store...
+             * (or in multiple stores if the test is in legacy mode)
+             * @param {String} storeName - the name of the sub store
+             * @returns {Promise<storage>}
+             */
+            getStore : function getStore(storeName) {
+                if(_.isEmpty(storeName)){
+                    throw new TypeError('A store name must be provided to get the store');
+                }
+
+                if(!_.contains(storeNames, storeName)){
+                    storeNames.push(storeName);
+                }
+                return getMode.then(function(){
```

Comment:
since `getMode` is not returning anything and has a side-effect (setting a 'global' var) then it could be renamed `setMode` (or changed to return the mode as a `resolve` parameter)